pkg/ansible/runner.go

```diff
@@ -0,0 +1,201 @@
+package ansible
+
+import (
+	"path"
+
+	log "github.com/sirupsen/logrus"
+
+	kbatch "k8s.io/api/batch/v1"
+	kapi "k8s.io/api/core/v1"
+	kapierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	openshiftAnsibleImage          = "openshift/origin-ansible:v3.7"
+	openshiftAnsibleServiceAccount = "cluster-operator-controller-manager"
+	inventoryConfigMap             = "ansible-inventory"
+	sshPrivateKeySecret            = "ssh-private-key"
+	operatorNamespace              = "cluster-operator"
+	openshiftAnsibleContainerDir   = "/usr/share/ansible/openshift-ansible/"
+	jobNamePrefix                  = "openshift-ansible-"
+	awsCredentialsSecret           = "aws-credentials"
+)
+
+type ansibleRunner struct {
+	KubeClient kubernetes.Interface
+	Image      string
+}
+
+func NewAnsibleRunner(kubeClient kubernetes.Interface) *ansibleRunner {
+	return &ansibleRunner{
+		KubeClient: kubeClient,
+		Image:      openshiftAnsibleImage,
+	}
+}
+func (r *ansibleRunner) createInventoryConfigMap(inventory string, vars string) (*kapi.ConfigMap, error) {
+	log.Debugln("Creating inventory/vars ConfigMap")
+	cfgmap := &kapi.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: inventoryConfigMap,
+		},
+		Data: map[string]string{
+			"hosts": inventory,
+			"vars":  vars,
+		},
+	}
+	cfgMap, err := r.KubeClient.CoreV1().ConfigMaps(operatorNamespace).Create(cfgmap)
+	if err != nil && kapierrors.IsAlreadyExists(err) {
+		// Update existing configmap if it already exists:
+		log.WithField("configmap", cfgMap.Name).Infoln("already exists, attempting update")
+		cfgMap, err = r.KubeClient.CoreV1().ConfigMaps(operatorNamespace).Update(cfgmap)
```

Comment:
I would be very cautious about updating en existing configmap unless we can be absolutely sure that there are no pods still using the configmap.

Reply:
Yeah this update block goes away once we're on one configmap per job.